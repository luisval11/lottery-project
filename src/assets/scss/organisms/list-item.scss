.list-item {
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
	border-radius: 6px;
	@include aboveBreakpoint() {
		padding-left: 21px;
	}
	padding-top: 21px;

	&__button {
		@include aboveBreakpoint() {
			margin-right: 21px;
			&--hidden {
				display: none;
			}
		}
	}

	&__image {
		background: linear-gradient(
			180deg,
			rgba(26, 23, 30, 0.64) 0%,
			rgba(26, 23, 30, 0) 24.98%,
			rgba(26, 23, 30, 0) 74.94%,
			rgba(26, 23, 30, 0.64) 100%
		);
		border-radius: 6px;
		@include aboveBreakpoint() {
			max-height: 96px;
			max-width: 132px;
		}
		@include belowBreakpoint() {
			max-height: 72px;
			max-width: 96px;
		}
		filter: drop-shadow(0px 5.33333px 26.6667px rgba(27, 24, 31, 0.12));
	}

	&__info {
		display: flex;
		flex-direction: column;
		align-items: flex-start;
		justify-content: flex-start;
		margin-left: 24px;
		margin-right: 24px;
	}

	&__info-image {
		position: absolute;
		left: 7px;
		top: 7px;
		z-index: 100;
		@include aboveBreakpoint() {
			max-height: 22px;
			max-width: 22px;
		}
		@include belowBreakpoint() {
			max-height: 14px;
			max-width: 15px;
		}

		&--hidden {
			position: absolute;
			left: 7px;
			top: 7px;
			z-index: 100;
			@include aboveBreakpoint() {
				max-height: 22px;
				max-width: 22px;
				visibility: hidden;
			}
			@include belowBreakpoint() {
				max-height: 14px;
				max-width: 15px;
			}
		}
	}

	&--selected {
		display: flex;
		flex-direction: row;
		align-items: center;
		justify-content: space-between;
		border-radius: 6px;
		@include aboveBreakpoint() {
			padding-left: 21px;
		}
		padding-top: 21px;
	}

	&__wrapper {
		position: relative;
		display: flex;
		align-items: center;
	}

	&__item-separator {
		@include aboveBreakpoint() {
			border-bottom: 1px solid rgba(204, 205, 217, 0.12);
			margin-top: 21px;
			margin-left: 23px;
		}
		@include belowBreakpoint() {
			border-bottom: 1px solid rgba(204, 205, 217, 0.12);
			margin-top: 14px;
		}
		&--selected {
			@include belowBreakpoint() {
				margin-top: 14px;
				border-bottom: 1px solid rgba(204, 205, 217, 0.12);
			}
		}
	}

	&__wrapper-rectangle {
		@include aboveBreakpoint() {
			min-height: 144px;
		}

		@include belowBreakpoint() {
			min-height: 107px;
		}

		&--selected {
			border-radius: 6px;
			@include aboveBreakpoint() {
				background: $color-casino-base;
				min-height: 144px;
			}
			@include belowBreakpoint() {
				min-height: 107px;
			}
		}
	}
}
