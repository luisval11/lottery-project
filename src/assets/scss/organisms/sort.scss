.sort {
	&__select {
		display: inline-block;
		background: $color-casino-base;
		color: $color-ui-casino-02;
		cursor: pointer;
		padding-bottom: 7px;
		padding-right: 17px;
		padding-top: 7px;
		border-radius: 4px;

		&-image {
			margin-left: 14px;
			margin-right: 7px;

			&--open {
				margin-left: 14px;
				margin-right: 7px;
				transform: scaleY(-1);
			}
		}

		&--open {
			display: inline-block;
			width: 272px;
			background: $color-casino-base;
			border-radius: 4px;
			color: $color-ui-casino-02;
			cursor: pointer;
			padding-bottom: 7px;
			padding-right: 17px;
			padding-top: 7px;
		}
	}

	&__wrapper-select {
		position: relative;
		width: fit-content;
		background: $color-casino-base;
		box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
		border-radius: 4px;
		display: flex;
		flex-direction: column;
		margin-left: 84px;
		margin-top: 25px;
		&--hidden {
			display: none;
		}
	}
}
