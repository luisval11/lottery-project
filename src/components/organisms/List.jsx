import { h } from "preact";
import { ListItem } from "../molecules/ListItem";

export const List = ({ items, className, handleSelectItem, selectedItem }) => {
	return (
		<div className={className}>
			{Object.values(items).map((item, id) => {
				return (
					<ListItem
						key={id}
						id={id}
						item={item}
						handleSelectItem={handleSelectItem}
						selectedItem={selectedItem}
					/>
				);
			})}
		</div>
	);
};
