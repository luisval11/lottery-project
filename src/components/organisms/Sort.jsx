import { Fragment, h } from "preact";
import chevronDown from "../../assets/images/chevron-down.png";
import { Image } from "../atoms/Image";
import { Radio } from "../molecules/Radio";

export const Sort = ({
	sortOptions,
	sortValue,
	isOpenSort,
	handleOpenSort,
	handleSelectOnChange,
	title,
}) => {
	return (
		<Fragment>
			<div className={`sort__wrapper-select${isOpenSort ? "--hidden" : ""}`}>
				<span className="sort__select" onClick={handleOpenSort}>
					<Image className="sort__select-image" src={chevronDown} />
					{title}
				</span>
			</div>
			<div className={`sort__wrapper-select${isOpenSort ? "" : "--hidden"}`}>
				<div>
					<span
						className={`sort__select${isOpenSort ? "--open" : ""}`}
						onClick={handleOpenSort}
					>
						<Image className="sort__select-image--open" src={chevronDown} />
						{title}
					</span>
				</div>
				<Radio
					value={sortValue}
					options={sortOptions}
					onChange={handleSelectOnChange}
				/>
			</div>
		</Fragment>
	);
};
