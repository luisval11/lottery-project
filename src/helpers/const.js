export const URL_LOTTOLAND = "https://www.lottoland.co.uk";
export const SORT_OPTION_NAME = "name";
export const SORT_OPTION_EUR_STAKE_ASC = "stakeEurAsc";
export const SORT_OPTION_EUR_STAKE_DESC = "stakeEurDesc";
export const MAX_GAMES_PER_PAGE = 9;
export const SORT_OPTIONS = [
	{ name: SORT_OPTION_NAME, text: "Name (A-Z)" },
	{ name: SORT_OPTION_EUR_STAKE_ASC, text: "Stake (min to max)" },
	{ name: SORT_OPTION_EUR_STAKE_DESC, text: "Stake (max to min)" },
];
export const EXAMPLE_LISTITEM = {
	LuckyWizard: {
		name: "LuckyWizard",
		displayName: "Lucky Wizard",
		playURL: "/games/luckywizard/play",
		image: "/cms/5bab9d950eb3580fac83392e/Icon_320and250_LuckyWizard.jpg",
		detailURL: "/games/luckywizard",
		currencyData: {
			CHF: {
				noDesk: true,
				noMob: true,
			},
			EUR: {
				minimumStake: 0.1,
			},
			GBP: {
				minimumStake: 0.1,
			},
			SEK: { minimumStake: 1.0 },
			PLN: {},
			JPY: {},
			INR: {},
			NZD: { minimumStake: 0.1 },
			CAD: {},
			HUF: {},
		},
		volatility: ["High"],
		provider: ["RedTiger"],
	},
	GreekGods: {
		name: "GreekGods",
		displayName: "Greek Gods",
		playURL: "/games/greekgods/play",
		image: "/cms/5de91d2a91253e002efe72f7/icon-320and250_GreekGods.jpg",
		detailURL: "/games/greekgods",
		currencyData: {
			EUR: { minimumStake: 0.25 },
			GBP: { minimumStake: 0.25 },
			SEK: { minimumStake: 2.5 },
			PLN: {},
			BRL: {},
			JPY: {},
			NZD: { minimumStake: 0.25 },
			INR: {},
			HUF: {},
		},
		volatility: ["High"],
		provider: ["Pragmatic"],
	},
};
