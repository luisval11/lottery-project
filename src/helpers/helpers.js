import { SORT_OPTION_EUR_STAKE_ASC, SORT_OPTION_EUR_STAKE_DESC, SORT_OPTION_NAME, URL_LOTTOLAND } from "./const";

export const normalizeUrl = (endpoint) => {
	return URL_LOTTOLAND + endpoint;
};

export const isValidGame = (game) => {
	return (
		game.currencyData != undefined &&
		game.currencyData.EUR != undefined &&
		game.currencyData.EUR.minimumStake != undefined
	);
};

export const getItemsToShow = (sort, initialValues, initial, size) => {
	switch (sort) {
		case SORT_OPTION_NAME:
			return Object.values(initialValues)
				.filter((game) => isValidGame(game))
				.sort((gameA, gameB) => {
					return gameA.displayName.localeCompare(gameB.displayName);
				})
				.slice(initial, size);

		case SORT_OPTION_EUR_STAKE_ASC:
			return Object.values(initialValues)
				.filter((game) => isValidGame(game))
				.sort((gameA, gameB) => {
					return (
						gameA.currencyData.EUR.minimumStake -
						gameB.currencyData.EUR.minimumStake
					);
				})
				.slice(initial, size);

		case SORT_OPTION_EUR_STAKE_DESC:
			return Object.values(initialValues)
				.filter((game) => isValidGame(game))
				.sort((gameA, gameB) => {
					return (
						gameB.currencyData.EUR.minimumStake -
						gameA.currencyData.EUR.minimumStake
					);
				})
				.slice(initial, size);
	}
};
