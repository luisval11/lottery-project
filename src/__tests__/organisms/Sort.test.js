import { h } from "preact";
import { mount, shallow } from "enzyme";
import { SORT_OPTIONS } from "../../helpers/const";
import { Sort } from "../../components/organisms/Sort";

describe("Test on Sort", () => {
	const sortTitle = "Sort Title";
	let sort = "name";
	let isOpenSort = false;
	const handleShowSortMenu = () => {
		isOpenSort = !isOpenSort;
	};
	const handleSelectOnChange = (value) => {
		sort = value;
	};

	test("should match initial snapshot", () => {
		let wrapper = shallow(
			<Sort
				title={sortTitle}
				sortOptions={SORT_OPTIONS}
				sortValue={sort}
				isOpenSort={isOpenSort}
				handleOpenSort={handleShowSortMenu}
				handleSelectOnChange={handleSelectOnChange}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	test("should 2 main divs, according to designs to Figma (div when menu is closed and div when is open)", () => {
		let wrapper = mount(
			<Sort
				title={sortTitle}
				sortOptions={SORT_OPTIONS}
				sortValue={sort}
				isOpenSort={isOpenSort}
				handleOpenSort={handleShowSortMenu}
				handleSelectOnChange={handleSelectOnChange}
			/>
		);
		expect(wrapper.find("div.sort__wrapper-select").length).toBe(1);
		expect(wrapper.find("div.sort__wrapper-select--hidden").length).toBe(1);
	});

	test("should have the text 'Sort Title'", () => {
		let wrapper = mount(
			<Sort
				title={sortTitle}
				sortOptions={SORT_OPTIONS}
				sortValue={sort}
				isOpenSort={isOpenSort}
				handleOpenSort={handleShowSortMenu}
				handleSelectOnChange={handleSelectOnChange}
			/>
		);
		expect(wrapper.find("span.sort__select").at(0).text()).toBe(sortTitle);
	});

	test("should have the image of 'chevron-down' to collapse or expand the menu", () => {
		let wrapper = mount(
			<Sort
				title={sortTitle}
				sortOptions={SORT_OPTIONS}
				sortValue={sort}
				isOpenSort={isOpenSort}
				handleOpenSort={handleShowSortMenu}
				handleSelectOnChange={handleSelectOnChange}
			/>
		);
		expect(wrapper.find("img.sort__select-image").length).toBe(1);
	});

	test("should have the menu of sorting collapsed when it's isOpenSort is closed", () => {
		let wrapper = mount(
			<Sort
				title={sortTitle}
				sortOptions={SORT_OPTIONS}
				sortValue={sort}
				isOpenSort={isOpenSort}
				handleOpenSort={handleShowSortMenu}
				handleSelectOnChange={handleSelectOnChange}
			/>
		);
		expect(wrapper.find("div.sort__wrapper-select--hidden").length).toBe(1);
	});

	test("if 2nd main div is hidden, when 1st main div is clicked, then hides the 1st one and 2nd main div is shown", () => {
		let wrapper = mount(
			<Sort
				title={sortTitle}
				sortOptions={SORT_OPTIONS}
				sortValue={sort}
				isOpenSort={isOpenSort}
				handleOpenSort={handleShowSortMenu}
				handleSelectOnChange={handleSelectOnChange}
			/>
		);

		//Extract the onClick for the div input
		const { onClick } = wrapper.find("span.sort__select").at(0).props();
		onClick();

		//Forces a rerender of the component with the new value after onClick;
		wrapper.setProps({
			title: sortTitle,
			sortOptions: SORT_OPTIONS,
			sortValue: sort,
			isOpenSort: isOpenSort,
			handleOpenSort: handleShowSortMenu,
			handleSelectOnChange: handleSelectOnChange,
		});
		wrapper.update();
		expect(wrapper.find("span.sort__select--open").length).toBe(1);
	});
});
