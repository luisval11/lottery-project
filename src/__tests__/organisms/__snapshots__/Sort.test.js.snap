// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Sort should match initial snapshot 1`] = `ShallowWrapper {}`;
